model_compression_toolkit/common/mixed_precision/kpi_methods.py

```diff
@@ -91,6 +89,48 @@ def activation_output_size_kpi(mp_cfg: List[int], graph: Graph, fw_info: Framewo
     return np.array(activation_memory)
 
 
+def total_weights_activation_kpi(mp_cfg: List[int], graph: Graph, fw_info: FrameworkInfo) -> np.ndarray:
+    """
+    Computes KPIs tensor with the respective weights size and output memory size for each activation configurable node,
+    according to the given mixed-precision configuration.
+
+    Args:
+        mp_cfg: A mixed-precision configuration (list of candidates index for each configurable node)
+        graph: Graph object.
+        fw_info: FrameworkInfo object about the specific framework (e.g., attributes of different layers' weights to quantize)
+            (not used in this method).
+
+    Returns: A 2D tensor of nodes' weights memory sizes and activation output memory size.
+    Note that the vector is not necessarily of the same length as the given config.
+
+    """
+    weights_activation_memory = []
+
+    # Go over all nodes that should be taken into consideration when computing the weights or
+    # activation KPI (all configurable nodes).
+    for node_idx, n in enumerate(graph.get_configurable_sorted_nodes()):
+        node_qc = n.candidates_quantization_cfg[mp_cfg[node_idx]]
+        node_weights_nbits = node_qc.weights_quantization_cfg.weights_n_bits
+        node_activation_nbits = node_qc.activation_quantization_cfg.activation_n_bits
+
+        # Compute node's weights memory (if no weights to quantize then set to 0)
+        node_num_weights_params = 0
+        for attr in fw_info.get_kernel_op_attributes(n.type):
+            if attr is not None:
+                node_num_weights_params += n.get_weights_by_keys(attr).flatten().shape[0]
+        node_weights_memory_in_bytes = node_num_weights_params * node_weights_nbits / BITS_TO_BYTES
+
+        # Compute node's activation memory (if node's activation are not being quantized then set to 0)
+        node_activation_memory_in_bytes = 0
+        if n.is_activation_quantization_enabled():
+            node_output_size = n.get_total_output_params()
+            node_activation_memory_in_bytes = node_output_size * node_activation_nbits / BITS_TO_BYTES
+
+        weights_activation_memory.append(np.array([node_weights_memory_in_bytes, node_activation_memory_in_bytes]))
+
+    return np.array(weights_activation_memory)
+
```

Comment:
Can't we use weight/activation functions and then merge them?

Reply:
Discussed offline